keps/sig-release/20200122-kubernetes-yearly-support-period.md

```diff
@@ -0,0 +1,318 @@
+  * Historically this trailing period outside of support was squishy ("wait a little while, cut a final patch release, turn down jobs").  It is currently specified that oldest release branch CI support is turned off at the time when the newest, under-development release branch is created.  This happens at the first ‘beta’ release, e.g.: ‘1.16.0-beta.0’.  Due to an implementation detail where branches are referred to by “dev”, “beta”, “stable1”, “stable2”, “stable3” there is a rotation of marker variables at the point of new branch creation.  This is an automated process, run by the SIG Release subproject Release Engineering branch manager, and can be adjusted as needed.  But while CI is running if a CVE or upgrade issue is encountered and resolution is able to be backported the community has intended to do so.  In practice this has not been required.
+  * This KEP proposes the rotation happen 2 months after the 1 year period, i.e.: 12+2 months patch ability.  This constitutes 12 months of normal patch releases, plus a two month grace period for end users to plan and upgrade their clusters to a supported version. During this special 2-month grace period only critical security and upgrade fixes are acceptable on the branch.  I.e.:
+    * Support Policy during the first 12 months.
+      * Same as the current 9 months policy.
+    * Support Policy during the final +2 months.
+      * Only critical security patches and upgrade blocking fixes, i.e.:
+        * CVE assigned by Product Security Committee initiated to release branch by Product Security Committee
+        * Cherry-pick of upgrade scenario bug fix approved by owning SIG and Patch Release Team
+* Formalize 3rd party dependency update policy (e.g.: golang, but also others too)
+    * Patch release updates: Need to track upstream golang patch releases continually.
+    * Major release updates: Can we also formalize when to move to a new Golang major release.  This is needed as two of our annual releases today (and all of them if we move to a yearly lifecycle) run beyond the end of Golang’s patch release lifecycle.
+### Clarify “Support Policy” in community documentation
+* Generally, e.g.:
+https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks
+* Specifically:
+  * 12 months general bug fix patching and backports
+  * 2 months critical security patches and upgrade blocking fixes
+
+### Document impact on external dependencies support windows
+
+* Example: golang support for ~1 year (2 6-month releases)
+* Options:
+  * Upgrade dependencies more on release branches
+  * Note where support gaps exist and create plans for more proactively managing deps
+
+### Test Plan
+
+N/A.
+The project already has pre-release test coverage and test automation.
+This KEP proposes no changes to this.
+Existing tests will continue running.
+
+### Graduation Criteria
+
+WG LTS proposes this KEP be implemented as of the 1.18.0 release.
```

Comment:
Agree 1.18 is too aggressive to get feedback and work through implications with all participating sigs. If the conversations start now then targeting 1.19 seems possible.

Reply:
I'll update, I've updated the description accordingly.